File: JavaServletSrc/LibreriaMit.java
import java.nio.charset.*;
import java.nio.CharBuffer;
import java.util.UUID;
import java.util.regex.Pattern;
import java.nio.ByteBuffer;
import java.net.URLEncoder;
import java.util.Random;
import java.util.Arrays;
public class LibreriaMit {

//---Start function: CalcCond
public static boolean CalcCond() {
  boolean res = true;
  return res;
}
//---End function

//---Start function: CloseLine
public static String CloseLine() {
  String res = CPLib.Chr(13)+CPLib.Chr(10);
  return res;
}
//---End function

//---Start function: FreeMem
public static void FreeMem() {
  System.gc();
//  System.out.println("Runtime reports "+Runtime.getRuntime().freeMemory()+" bytes free.");
}
//---End function

//---Start function: GeneraCodice
public static String GeneraCodice(int numeroCaratteriRandom, boolean conSpeciali, boolean conTimeStamp, String separatore) {
  // array delle lettere
  String[] Caratteri = { "a", "b", "c", "d", "e", "f", "g", "h", "i",
	  "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "z", "y",
		"j", "k", "x", "w", "A", "B", "C", "D", "E", "F", "G", "H", "I",
		"L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "Z", "Y",
		"J", "K", "X", "W", "à", "è", "é", "ì", "ò", "ù" };
  // array dei numeri
  String[] Numeri = { "1", "2", "3", "4", "5", "6", "7", "8", "9", "0" };
  // array dei caratteri speciali
  String[] Speciali = {"!","£","$","%","&","@","*",",","_","-","#",";","^",":", ".", "+", "§", "?", "ç","/"};  
  // creo l'oggetto random
	Random rand = new Random();
  // ottengo la lunghezza di ogni array
	int lunghezzaCaratteri = Caratteri.length;
	int lunghezzaNumeri = Numeri.length;
	int lunghezzaSpeciali = Speciali.length;
	// istanzio la variabile che conterrà il prodotto finale
	String stringaRandom = "";
	while (stringaRandom.length() < numeroCaratteriRandom) {
		// ottengo un elemento casuale per ogni array
		int c = rand.nextInt(lunghezzaCaratteri);
		int n = rand.nextInt(lunghezzaNumeri);
		int s = rand.nextInt(lunghezzaSpeciali);
		// aggiungo una lettera casuale
		stringaRandom += Caratteri[c];
		// aggiungo un numero random
		stringaRandom += Numeri[n];
		// se l'opzione conSpeciali è true aggiungo un carattere speciale
		if (conSpeciali) {
			stringaRandom += Speciali[s];
		}
	}
	// se la stringa generata dovesse superare il numero di caratteri
	// richiesto, la taglio.
	if (stringaRandom.length() > numeroCaratteriRandom) {
		stringaRandom = stringaRandom.substring(0, numeroCaratteriRandom);
	}
	// se abbiamo deciso di mettere il timestamp
	if (conTimeStamp) {
		// recupero il timestamp
		long timestamp = System.currentTimeMillis();
		stringaRandom += separatore + timestamp;
	}
	// restituisco la stringa generata
	return stringaRandom;
}
//---End function

//---Start function: NewStato
public static String NewStato(String cTbRif, String cFlag) {
  String res = "!";
  String _oper = "ABCDEFGHI";
  String _fraz = "JLMNOTUV?";
  if (cTbRif == "OPERAZBO") {
    for (int i=1;i<=9;i++) {
      if (i == CPLib.Val(cFlag)) {
        res = CPLib.Substr(_oper,i,1);
      }
    }
  }
  else {
    for (int i=1;i<=9;i++) {
      if (i == CPLib.Val(cFlag)) {
        res = CPLib.Substr(_fraz,i,1);
      }
    }    
  }  
  return res;
}
//---End function

//---Start function: Occurs
public static double Occurs(String cOrigin, String cSearch) {
  double result = 0;
  int len = cSearch.length();
	if (len > 0) {  
	  int start = cOrigin.indexOf(cSearch);
		while (start != -1) {
				result++;
				start = cOrigin.indexOf(cSearch, start+len);
	  }
  }
  return result;
}
//---End function

//---Start function: SpacePad
public static String SpacePad(String cStringa, double nLen) {
  String res = "";
  res = CPLib.Left(CPLib.LRTrim(CPLib.Left(cStringa,nLen))+CPLib.Space(nLen),nLen);
  return res;
}
//---End function

//---Start function: StatoReg
public static String StatoReg(String cTbRif, String nStato) {
  String res = nStato;
//  if (cTbRif == 'RAPOPEBO') {
//    if (nStato == '1') {
//      res = 'M'
//    }  
//    else if (nStato == '2') {
//      res = 'C'
//    }  
//    else if (nStato == '3') {
//      res = 'V'
//    }  
//    else {
//      res = 'B'
//    }
//    res = nStato
//  }
  if (cTbRif == "SIMINFBO") {
    if (nStato == "1") {
      res = "M";
    }  
    else if (nStato == "2") {
      res = "C";
    }  
    else if (nStato == "3") {
      res = "V";
    }  
    else {
      res = "B";
    }
  }
  if (cTbRif == "OPERAZBO") {
    if (nStato == "1") {
      res = "P";
    }  
    else if (nStato == "2") {
      res = "Q";
    }  
    else if (nStato == "3") {
      res = "R";
    }  
    else {
      res = "S";
    }
  }
  if (cTbRif == "FRAZIOBO") {
    if (nStato == "1") {
      res = "K";
    }  
    else if (nStato == "2") {
      res = "X";
    }  
    else if (nStato == "3") {
      res = "Y";
    }  
    else {
      res = "Z";
    }
  }  
  if (cTbRif == "SIMOPEBO") {
    if (nStato == "1") {
      res = "P";
    }  
    else if (nStato == "2") {
      res = "Q";
    }  
    else if (nStato == "3") {
      res = "R";
    }  
    else {
      res = "S";
    }
  }
  return res;
}  
//---End function

//---Start function: TestAscii
public static boolean TestAscii(String sStringa, String sCharSet) {
  boolean res;
  byte bytearray []  = sStringa.getBytes();
  CharsetDecoder d = Charset.forName(sCharSet).newDecoder();
  try {
       CharBuffer r = d.decode(ByteBuffer.wrap(bytearray));
       res = true;
  }
  catch(CharacterCodingException e) {
       res = false;
  }  
  return res;
}
//---End function

//---Start function: URL_enc
public static String URL_enc(String pValue) {
  String value;
  value=pValue;
  try {
    value=URLEncoder.encode(pValue,com.zucchetti.sitepainter.Library.m_cResponseEncoding);
  } catch(Exception couldntEncode) {
    CPStdCounter.Error(couldntEncode);
  }
  return value;
}
//---End function

//---Start function: UTFtoISO
public static String UTFtoISO(String s1) {
  String s = new String(s1.getBytes(StandardCharsets.UTF_8));
  byte[] b = s.getBytes(StandardCharsets.ISO_8859_1);
  return new String(b, StandardCharsets.ISO_8859_1);
}
//---End function

//---Start function: UniqueId
public static String UniqueId() {
  char[] cpccchk = new char[10];
  for (int i=0; i<10; i++)
    cpccchk[i]=(char)(Math.random()*26+97);
  return new String(cpccchk);
}
//---End function

//---Start function: cWarnOpe
public static String cWarnOpe(String cErrMsg, String cTipCon) {
  String res = "";
  return res;
}
//---End function

//---Start function: cWarning
public static String cWarning(String cErrMsg) {
  String res = "";
  return res;
}
//---End function

//---Start function: inter_art1
public static boolean inter_art1(String pTipo) {
  boolean res = false;
  if(CPLib.At(pTipo,"|02|03|04|06|07|11|16|18|46|47|49|81|82")>0)
    res=true;
  return res;
}
//---End function

//---Start function: isAlpha
public static boolean isAlpha(String pString) {
  boolean res = false;
  Pattern p = Pattern.compile("[a-zA-Z]+");
  res = p.matcher(pString).find();
  return res;
}
//---End function

//---Start function: isAlphaNum
public static boolean isAlphaNum(String pStringa) {
  boolean res = false;
  Pattern p = Pattern.compile("[a-zA-Z0-9]+");
  res = p.matcher(pStringa).find();
  return res;
}
//---End function

//---Start function: isNumeric
public static boolean isNumeric(String pString) {
  boolean res = false;
  String regex="\\d+";
  res = pString.matches(regex);
  return res;
}
//---End function

//---Start function: nextUID
public static String nextUID() {
  UUID uuid = UUID.randomUUID();
  String res = uuid.toString();
  return res;
}
//---End function

//---Start function: uuidToStr15
public static String uuidToStr15(UUID uuid) {
  long[] longs = new long[2];
  longs[0] = uuid.getLeastSignificantBits();
  longs[1] = uuid.getMostSignificantBits();
  System.out.println("uuidToStr15: " + Arrays.toString(longs));
  char[] chars = new char[15];
  // 15 chars x 9 bits payload == 135 >=  128.
  final int bitsPerChar = (128 + chars.length - 1) / chars.length;
  final int char0 = 0x2000;
  long mask = (1L << bitsPerChar) - 1;
  for (int i = 0; i < chars.length; ++i) {
    int payload = (int)(longs[0] & mask);
    chars[i] = (char)(char0 + payload);
    longs[0] >>>= bitsPerChar;
    longs[0] |= (longs[1] & mask) << (64 - bitsPerChar);
    longs[1] >>>= bitsPerChar;
  }
  System.out.println(chars);
  return new String(chars);
}
//---End function

//---Start attributes
//---End attributes
}

File: Libraries/LibreriaMit/JavaServlet/LibreriaMit.js
(function() {//MakeLibreriaMit
function ReadVisualQuery(cCmd, cParam)
{
  // Valorizzazione dei parametri
  // cCmd: Nome della Visual Query da eseguire (Senza Suffisso)
  // CParam: Stringa Elenco dei parametri da passare alla query
  // Indicare senza w_;
  // se più di 1 separati da virgola; ''=Nessuno
  var strparms='';
  // Interpreta i Parametri per la VQuery
  if(!Empty(cParam)) {
    var aparms=cParam.split(',');
    for(var i=0;i<aparms.length;i++) {
      if(Trim(aparms[i])!='') {
        //prende i valori dai relativi textbox
        get_par=URLenc(aparms[i]);
        strparms+='&w_'+get_par+'='+CtrlByName(get_par).value;
      }
    }
  }
  // Istanzia l'Oggetto per la gestione delle richieste al Server
  var url = new JSURL('../servlet/SQLDataProviderServer?rows=10000&startrow=0&sqlcmd='+URLenc(cCmd)+strparms,true);
  // Invia la richiesta al Server (Ritorna un Array)
  var output=url.Response();
  var Data = eval(output);
  //Prende il secondo elelemto dell'array, dal momento che il primo elemento
  //contiene la struttura della risposta della query
  Data=Data.pop();
  //Eliminiamo l'ultimo elemento dell'array che contiene dei codici di controllo
  var dummy = Data.pop();
  //Ritorna: Data=Array bidimensionale contenente cursore query
  //Data[Numero Record][Posizione Campo]
  //Data.length: Ritorna il numero di record letti dalla query. 
  return Data;
} 


//---Start function: CalcCond
function libreriamit_CalcCond(oVal,xVal) {
  if (Empty(oVal) && ! Empty(xVal)){
    w_attsel = w_attsel + 1    
  }  
  if (Empty(xVal) && ! Empty(oVal)) {
    w_attsel = w_attsel - 1
  }
}
//---End function

//---Start function: CloseLine
function libreriamit_CloseLine() {
  var res = Chr(13)+Chr(10);
  return res;
}
//---End function

//---Start function: FreeMem
function libreriamit_FreeMem() {
}
//---End function

//---Start function: GeneraCodice
function libreriamit_GeneraCodice(numeroCaratteriRandom, conSpeciali, conTimeStamp, separatore) {
  var res = "";
  return res;
}
//---End function

//---Start function: NewStato
function libreriamit_NewStato(cTbRif, cFlag) {
  var res = "!";
  var _oper = "ABCDEFGHI"
  var _fraz = "JLMNOTUV?"
  if (cTbRif == 'OPERAZBO') {
    for (i=1;i<=9;i++) {
      if (i == Val(cFlag)) {
        res = Substr(_oper,i,1)
      }
    }
  }
  else {
    for (i=1;i<=9;i++) {
      if (i == Val(cFlag)) {
        res = Substr(_fraz,i,1)
      }
    }    
  }  
  return res;
}
//---End function

//---Start function: Occurs
function libreriamit_Occurs(cOrigin, cSearch) {
  var result = 0;
  var len = cSearch.length;
	if (len > 0) {  
	  var start = cOrigin.indexOf(cSearch);
		while (start != -1) {
				result++;
				start = cOrigin.indexOf(cSearch, start+len);
	  }
  }
  return result;
}
//---End function

//---Start function: SpacePad
function libreriamit_SpacePad(cStringa, nLen) {
  var res = "";
  res = res = Left(LRTrim(Left(cStringa,nLen))+Space(nLen),nLen);
  return res;
}
//---End function

//---Start function: StatoReg
function libreriamit_StatoReg(cTbRif, nStato) {
  var res = nStato;
//  if (cTbRif == 'RAPOPEBO') {
//    if (nStato == '1') {
//      res = 'M'
//    }  
//    else if (nStato == '2') {
//      res = 'C'
//    }  
//    else if (nStato == '3') {
//      res = 'V'
//    }  
//    else {
//      res = 'B'
//    }
//    res = nStato
//  }
  if (cTbRif == 'SIMINFBO') {
    if (nStato == '1') {
      res = 'M'
    }  
    else if (nStato == '2') {
      res = 'C'
    }  
    else if (nStato == '3') {
      res = 'V'
    }  
    else {
      res = 'B'
    }
  }
  if (cTbRif == 'OPERAZBO') {
    if (nStato == '1') {
      res = 'P'
    }  
    else if (nStato == '2') {
      res = 'Q'
    }  
    else if (nStato == '3') {
      res = 'R'
    }  
    else {
      res = 'S'
    }
  }
  if (cTbRif == 'FRAZIOBO') {
    if (nStato == '1') {
      res = 'K'
    }  
    else if (nStato == '2') {
      res = 'X'
    }  
    else if (nStato == '3') {
      res = 'Y'
    }  
    else {
      res = 'Z'
    }
  }  
  if (cTbRif == 'SIMOPEBO') {
    if (nStato == '1') {
      res = 'P'
    }  
    else if (nStato == '2') {
      res = 'Q'
    }  
    else if (nStato == '3') {
      res = 'R'
    }  
    else {
      res = 'S'
    }
  }
  return res;
}
//---End function

//---Start function: TestAscii
function libreriamit_TestAscii(sStringa, sCharSet) {
  var res = false;
  return res;
}
//---End function

//---Start function: URL_enc
function libreriamit_URL_enc(pValue) {
  var e=Lower(document.charset)
  return Strtran(Strtran(escape(pValue),"+","%2B"),"%u20AC",e=='utf-8'?"%E2%82%AC":(e=='iso-8859-15'?"%A4":"%80"))
}
//---End function

//---Start function: UTFtoISO
function libreriamit_UTFtoISO(s1) {
  var res = "";
  return res;
}
//---End function

//---Start function: UniqueId
function libreriamit_UniqueId() {
  var res = ""
  for (var i=0; i<10; i++) {
     res=res + String.fromCharCode(Math.random()*26+97)
  }   
  return res
}
//---End function

//---Start function: cWarnOpe
function libreriamit_cWarnOpe(cErrMsg, cTipCon) {  var res = "";  if (Empty(cErrMsg)){}  else { alert(cErrMsg)  }  if(cTipCon == "S" || cTipCon == 'N' || cTipCon == ' ')  { res="P"  }  else  { res="E"  }  return res;  }
//---End function

//---Start function: cWarning
function libreriamit_cWarning(cErrMsg) {  var res = "";  if (Empty(cErrMsg)){}  else { alert(cErrMsg)  }  return res;  }
//---End function

//---Start function: inter_art1
function libreriamit_inter_art1(pTipo) {
  var res = false;
  if(At(pTipo,"|02|03|04|06|07|11|16|18|46|47|49|81|92")>0)
    res=true;
  return res;
}
//---End function

//---Start function: isAlpha
function libreriamit_isAlpha(pString) {
  var res = false;
  return res;
}
//---End function

//---Start function: isAlphaNum
function libreriamit_isAlphaNum(pStringa) {
  var res = false;
  return res;
}
//---End function

//---Start function: isNumeric
function libreriamit_isNumeric(pString) {
  var res = false;
  return res;
}
//---End function

//---Start function: nextUID
function libreriamit_nextUID() {
  var res = "";
  return res;
}
//---End function

//---Start function: uuidToStr15
function libreriamit_uuidToStr15(uuid) {
  var res = "";
  return res;
}
//---End function

//---Start function: MakeLibreriaMit
function MakeLibreriaMit() {
  this.CalcCond=libreriamit_CalcCond
  this.CloseLine=libreriamit_CloseLine
  this.FreeMem=libreriamit_FreeMem
  this.GeneraCodice=libreriamit_GeneraCodice
  this.NewStato=libreriamit_NewStato
  this.Occurs=libreriamit_Occurs
  this.SpacePad=libreriamit_SpacePad
  this.StatoReg=libreriamit_StatoReg
  this.TestAscii=libreriamit_TestAscii
  this.URL_enc=libreriamit_URL_enc
  this.UTFtoISO=libreriamit_UTFtoISO
  this.UniqueId=libreriamit_UniqueId
  this.cWarnOpe=libreriamit_cWarnOpe
  this.cWarning=libreriamit_cWarning
  this.inter_art1=libreriamit_inter_art1
  this.isAlpha=libreriamit_isAlpha
  this.isAlphaNum=libreriamit_isAlphaNum
  this.isNumeric=libreriamit_isNumeric
  this.nextUID=libreriamit_nextUID
  this.uuidToStr15=libreriamit_uuidToStr15
}
//---End function

arguments[0].LibreriaMit = new MakeLibreriaMit()
})(window);//MakeLibreriaMit
